src/main/java/net/minestom/server/instance/Chunk.java

```diff
@@ -399,28 +406,19 @@ public ChunkDataPacket getFreshPartialDataPacket() {
      */
     @NotNull
     public UpdateLightPacket getLightPacket() {
-        // TODO do not hardcode light
-        UpdateLightPacket updateLightPacket = new UpdateLightPacket(getIdentifier(), getLastChangeTime());
-        updateLightPacket.chunkX = getChunkX();
-        updateLightPacket.chunkZ = getChunkZ();
-        updateLightPacket.skyLightMask = 0x3FFF0;
-        updateLightPacket.blockLightMask = 0x3F;
-        updateLightPacket.emptySkyLightMask = 0x0F;
-        updateLightPacket.emptyBlockLightMask = 0x3FFC0;
-        byte[] bytes = new byte[2048];
-        Arrays.fill(bytes, (byte) 0xFF);
-        List<byte[]> temp = new ArrayList<>(14);
-        List<byte[]> temp2 = new ArrayList<>(6);
-        for (int i = 0; i < 14; ++i) {
-            temp.add(bytes);
-        }
-        for (int i = 0; i < 6; ++i) {
-            temp2.add(bytes);
-        }
-        updateLightPacket.skyLight = temp;
-        updateLightPacket.blockLight = temp2;
+        ChunkLight lightData = new ChunkLight(this);
+        chunkLightEngine.lightChunk(lightData);
```

Comment:
Light should not be calculated when the packet is needed, it should be completely generated when the chunk is generated, and then only updated when lights are added or removed.

Reply:
As stated in the cache comment, I'll be sure to make the adjustment.